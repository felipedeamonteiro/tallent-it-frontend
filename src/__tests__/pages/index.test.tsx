import React from 'react';
import { render } from '@testing-library/react';
import Home from '../../pages/index';
import '@testing-library/jest-dom';
import MockAdapter from 'axios-mock-adapter';
import api from '../../services/api';

const apiMock = new MockAdapter(api);

describe('Home', () => {
  it('should render the Homepage', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  });

  it('should be able to fetch data and show on the screen', () => {
    const apiResponse = {
      id: 1,
      name: "Leanne Graham",
      username: "Bret",
      email: "Sincere@april.biz",
      address: {
        street: "Kulas Light",
        suite: "Apt. 556",
        city: "Gwenborough",
        zipcode: "92998-3874",
        geo: {
          lat: "-37.3159",
          lng: "81.1496"
        }
      },
    };

    apiMock.onGet('users').reply(200, apiResponse);

  });
});
