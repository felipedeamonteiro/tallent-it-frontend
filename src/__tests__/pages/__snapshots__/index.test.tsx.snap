// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a the Homepage 1`] = `
<div>
  <div
    class="c-kXlFrX"
  >
    <h1
      class="page-title"
    >
      Tallent IT Frontend Test
    </h1>
    <button
      class="c-fbJfXZ"
    >
      Click to get data
    </button>
  </div>
</div>
`;
